transactId-library-ios/transactId-library-ios/Bip75/Bip75ServiceNetki.swift

```diff
@@ -264,7 +248,135 @@ class Bip75ServiceNetki: Bip75Service {
+                        let isSenderEncryptionSignatureValid = try paymentBinary.validateMessageEncryptionSignature()
+                        if (!isSenderEncryptionSignatureValid) {
+                            throw Exception.InvalidSignatureException(ExceptionMessages.signatureValidationInvalidSenderSignature)
+                        }
+                    }
+                    
+                    try messagePayment.originators.forEach { (messageOriginator) in
+                        try messageOriginator.attestations.forEach { (messageAttestation) in
+                            
+                            let isCertificateOwnerChainValid = try validateCertificate(pkiType: messageAttestation.getMessagePkiType(), certificate: messageAttestation.pkiData.toString())
+                            
+                            if (!isCertificateOwnerChainValid) {
+                                throw Exception.InvalidCertificateChainException(ExceptionMessages.certificateValidationInvalidOwnerCertificateCA)
+                            }
+                            
+                            let isSignatureValid = try messageAttestation.validateMessageSignature(requireSignature: messageOriginator.isPrimaryForTransaction)
+                            
+                            if (!isSignatureValid) {
+                                throw Exception.InvalidSignatureException(ExceptionMessages.signatureValidationInvalidOwnerSignature)
+                            }
+                        }
+                    }
+                    
+                    try messagePayment.beneficiaries.forEach { (messageBeneficiary) in
+                        try messageBeneficiary.attestations.forEach { (messageAttestation) in
+                            let isCertificateOwnerChainValid = try self.validateCertificate(pkiType: messageAttestation.getMessagePkiType(), certificate: messageAttestation.pkiData.toString())
+                            if (!isCertificateOwnerChainValid) {
+                                throw Exception.InvalidCertificateChainException(ExceptionMessages.certificateValidationInvalidOwnerCertificateCA)
+                            }
+                        }
+                    }
+                    return true
+                }
+            }
+        }
+        return false
+    }
+    
+    func parsePayment(paymentBinary: Data, recipientParameters: RecipientParameters?) throws -> Payment? {
+        if let protocolMessageMetadata = try paymentBinary.extractProtocolMessageMetadata() {
+            if let messagePaymentData = try paymentBinary.getSerializedMessage(encrypted: protocolMessageMetadata.encrypted, recipientParameters: recipientParameters) {
+                if let messagePayment = try messagePaymentData.toMessagePayment() {
+                    return try messagePayment.toPayment(protocolMessageMetadata: protocolMessageMetadata)
+                }
+            }
+        }
+        return nil
```

Comment:
*Not sure how swift works.
Instead of returning nil, should we throw an exception that we could not parse?

Reply:
This code will never be executed if we have some exception. We have to return nil here because we have to return something.